### ISSUE TITLE
White bar atop the game windows

### ISSUE THREAD

**Issue by Susu357** [KC3Kai/KC3Kai#3104]
In the recent update, I found out the game seemed a little bit off. 

There's a small white bar formed atop the game windows. It completely ruins the whole gameplay experience as it drags the windows down.

How to fix this?
![Capture](https://user-images.githubusercontent.com/81631956/112998781-f23fe280-91a0-11eb-8c79-db4630a044d8.JPG)

**Comment by sinsinpub** [comment]
I have no idea at all. should not be related to kc3 recent update.
a possibility I guess: some other extensions you installed have injected some elements/styles in to your dmm page, which push your game window off.

**Comment by jimmyzzxhlh** [comment]
Have the exact same problem, but it seems to be server specific. I play the game in Lingga, Sukumo, and Paramushir servers. Both Lingga and Paramushir have this problem but not Sukumo.

It looks like there is an empty `spacing_top` div with 16px height added. You can just remove the div from the console but it has to be done every time when the game is loaded. 

![image](https://user-images.githubusercontent.com/11505867/216773379-4ea8c0c1-ac45-4923-9711-5a4c8b052958.png)

Sukumo has the div as well but it's 0px height.

![image](https://user-images.githubusercontent.com/11505867/216773733-dd6a63fa-2871-4afe-a557-b4a8194f296a.png)

**Comment by sinsinpub** [comment]
i don't think it's server specific tho, since kc3 hide `spacing_top` in the `osapi.dmm.com` sub frame page, irrelevant to game server. u can check codes at https://github.com/KC3Kai/KC3Kai/blob/master/src/library/injections/osapi.js#L51

but one possibility is: if the injection of scripts fails for those server pages? can u show me the full url link of `osapi.dmm.com` iframe src? (just scrolling the elements up to find)
or u can check if osapi injection works by changing game player scale to a different zoom value in kc3 settings. osapi page will not be correctly resized if injection failed for those servers

**Comment by jimmyzzxhlh** [comment]
Sure this is Lingga:

![image](https://user-images.githubusercontent.com/11505867/216777104-9644dd69-1268-4b08-ab8d-69a07516d927.png)

Source is (hiding the token just to be safe :) ):

```
http://osapi.dmm.com/gadgets/ifr?synd=dmm&container=dmm&owner=18732953&viewer=18732953&aid=854854&mid=39634579&country=jp&lang=ja&view=canvas&parent=%2F%2Fwww.dmm.com%2Fnetgame%2Fsocial%2F&url=http%3A%2F%2F203.104.209.7%2Fgadget_html5.xml&st=xxx#rpctoken=yyy
```

And this is Sukumo:

![image](https://user-images.githubusercontent.com/11505867/216777241-756732c5-9a09-44da-a518-89cde98e7a03.png)

```
http://osapi.dmm.com/gadgets/ifr?synd=dmm&container=dmm&owner=13874561&viewer=13874561&aid=854854&mid=18134439&country=jp&lang=ja&view=canvas&parent=%2F%2Fwww.dmm.com%2Fnetgame%2Fsocial%2F&url=http%3A%2F%2F203.104.209.7%2Fgadget_html5.xml&st=xxx#rpctoken=yyy
```

**Comment by jimmyzzxhlh** [comment]
Tried to change the scale from 100 to 80 and the bar still exists.

**Comment by sinsinpub** [comment]
injection seems be working for both... since `zoom: 1` added to body?

alternatively try to use 'DMM frame' mode to play on Lingga?
![image](https://user-images.githubusercontent.com/160240/216777839-857d35da-8813-4f60-afcd-0350fb321ff0.png)

**Comment by jimmyzzxhlh** [comment]
Yeah dmm frame mode works OK, although I can live with the white bar as well.
Code logic in osapi.js looks good to me, and manually executing `$("#spacing_top").hide();` works as well, not sure what's going on here...

**Comment by sinsinpub** [comment]
then i've no idea too... not sure if the execution broken by different browser versions?
could u possible to set a breakpoint to check if that line actually executed?

**Comment by jimmyzzxhlh** [comment]
So the `.hide()` and the `hideSpacingTop` function are not executed because the `response.storage[0]` is false instead of true for Lingga. It is set to true for Sukumo. Following is from Lingga. 

![image](https://user-images.githubusercontent.com/11505867/216783856-6c360d72-1cbe-4ef3-9c08-6dc1d98222fc.png)

What controls the `response.storage[0]` here?

**Comment by sinsinpub** [comment]
`.storage[0]: true` stands for 'dmm site' play mode; `false` means 'dmm frame' mode or 'api link' mode (`[1]=='true'` in this case).
so r u sure that two false values gotten even when it was started in dmm site mode?

**Comment by jimmyzzxhlh** [comment]
> probably u were watching dmm frame mode values?

I don't use dmm frame mode normally (only tried it out when writing one of the above comments). In all 3 accounts, I turned on the `Apply Customizations` toggle, so that I can get the benefit of exit confirmation. The Game Modes is set to the default `Customize DMM Website` which I believe it is the "dmm site play mode" that you mentioned. 

If I understand the code correctly, turning on `Apply Customizations` should set the `storage[0]` to true. Is that correct? Right now I don't see that being the case for my Lingga account. I tried turning `Apply Customizations` off and then on, but both values in `storage` are always set to false.

**Comment by sinsinpub** [comment]
nope, turning on `Apply Customizations` set `.value[1]: true`, irrelevant to `storage`.
ofc `Apply Customizations` off means no change of dmm website page's looks&feels, so no spacing top hiding either.

if u were inspecting `storage[0]='false'` when playing via default mode, it means `localStorage` value not correctly passed from 'background scripts' to 'content scripts'. no considerable reason for this, except browser version change, since there are working instances...

i don't have such buggy browser at my side, can't help to find a solution for now if no more details figured out. so suggest to use dmm frame mode alternately, should be okay for gameplay mostly (not sure in-game purchase works or not since it depends on dmm platform)

**Comment by sinsinpub** [comment]
well, just thinking, if the issue is on data passing of parameters, i could modify dmm frame mode to use the same method to hide spacing top div (always hide it despite of configs?)

btw, legacy dmm frame mode hide spacing top by moving the whole subframe top position -16px because the div is 16px height. it's definitely worse method if devs change the height some day (although it's untouched for over 8 years in fact)

**Comment by jimmyzzxhlh** [comment]
>  i could modify dmm frame mode to use the same method to hide spacing top div (always hide it despite of configs?)

Yeah I guess hiding the spacing top div regardless of the config would probably work (and probably is also cleaner than moving the position by -16px :) )

**Comment by jimmyzzxhlh** [comment]
> if u were inspecting storage[0]='false' when playing via default mode, it means localStorage value not correctly passed from 'background scripts' to 'content scripts'.

I just checked the value in local storage for `dmmplay` and it was set to false. Setting to true manually solves the problem.

![image](https://user-images.githubusercontent.com/11505867/216834245-5d2f92b9-5a8b-43bf-b2a7-865f226e408e.png)

**Comment by sinsinpub** [comment]
for these environment, try to set direct boolean value to `localStorage` under any Strategy Room page console (the same extension context), eg:
```
localStorage.dmmplay = true
```
to check if browser only accepts strings on `localStorage` object?

**Comment by jimmyzzxhlh** [comment]
That seems to be working fine. The `dmmplay` is set in local storage.

![image](https://user-images.githubusercontent.com/11505867/216836111-5280a516-e1d7-4239-a8cd-57e897457912.png)
![image](https://user-images.githubusercontent.com/11505867/216836145-bfcd25d6-2a3f-4bf0-98b5-c9147178390c.png)
